server2/Arcadia.Assistant/Arcadia.Assistant.Sharepoint/Sharepoint.cs

```diff
@@ -0,0 +1,307 @@
+
+    using SickLeaves.Contracts;
+
+    using Vacations.Contracts;
+
+    using WorkHoursCredit.Contracts;
+
+    /// <summary>
+    ///     An instance of this class is created for each service instance by the Service Fabric runtime.
+    /// </summary>
+    public class Sharepoint : StatelessService
+    {
+        private readonly ISharepointDepartmentsCalendarsSettings departmentsCalendarsSettings;
+        private readonly IEmployees employees;
+        private readonly Func<IExternalStorage> externalStorageProvider;
+        private readonly IOrganization organizations;
+        private readonly ISharepointSynchronizationSettings serviceSettings;
+        private readonly IEqualityComparer<StorageItem> sharepointStorageItemComparer = new SharepointStorageItemComparer();
+        private readonly ISickLeaves sickLeaves;
+        private readonly IVacations vacations;
+        private readonly IWorkHoursCredit workouts;
+
+        public Sharepoint(
+            StatelessServiceContext context,
+            IVacations vacations,
+            IWorkHoursCredit workouts,
+            ISickLeaves sickLeaves,
+            IEmployees employees,
+            IOrganization organizations,
+            Func<IExternalStorage> externalStorageProvider,
+            ISharepointSynchronizationSettings serviceSettings,
+            ISharepointDepartmentsCalendarsSettings departmentsCalendarsSettings)
+            : base(context)
+        {
+            this.externalStorageProvider = externalStorageProvider;
+            this.vacations = vacations;
+            this.workouts = workouts;
+            this.sickLeaves = sickLeaves;
+            this.employees = employees;
+            this.organizations = organizations;
+            this.serviceSettings = serviceSettings;
+            this.departmentsCalendarsSettings = departmentsCalendarsSettings;
+        }
+
+        /// <summary>
+        ///     Optional override to create listeners (e.g., TCP, HTTP) for this service replica to handle client or user requests.
+        /// </summary>
+        /// <returns>A collection of listeners.</returns>
+        protected override IEnumerable<ServiceInstanceListener> CreateServiceInstanceListeners()
+        {
+            return new ServiceInstanceListener[0];
+        }
+
+        /// <summary>
+        ///     This is the main entry point for your service instance.
+        /// </summary>
+        /// <param name="cancellationToken">Canceled when Service Fabric needs to shut down this service instance.</param>
+        protected override async Task RunAsync(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                //ServiceEventSource.Current.ServiceMessage(this.Context, "Working-{0}", ++iterations);
+
+                // request Sharepoint calendars
+                var externalStorage = this.externalStorageProvider();
```

Comment:
yeah, please put everything from this loop statement into a separate class :)

Reply:
Excluded